app/components/views/Stake/index.tsx

```diff
@@ -100,17 +115,33 @@ export const Stake = (props: Props) => {
   const handleApproval = (action: "stake" | "unstake") => async () => {
     if (!props.provider) return;
     try {
-      const value = await changeApprovalTransaction({
+      const currentQuantity = quantity.toString();
+      const approvedValue = await changeApprovalTransaction({
+        value: currentQuantity,
         provider: props.provider,
         action,
         onStatus: setStatus,
       });
+
       if (action === "stake") {
-        dispatch(setStakeAllowance({ klima: value }));
+        dispatch(
+          incrementAllowance({
+            token: "klima",
+            spender: "staking_helper",
+            value: approvedValue,
+          })
+        );
       } else {
-        dispatch(setStakeAllowance({ sklima: value }));
+        dispatch(
+          incrementAllowance({
+            token: "sklima",
+            spender: "staking",
```

Comment:
love this!

Reply:
I guess as far as naming, now the actions should just be called `setAllowance` and `decrementAllowance`
we never actually add two values together (increment)

Reply:
Done